src/coreclr/jit/codegenxarch.cpp

```diff
@@ -3546,8 +3553,15 @@ void CodeGen::genCodeForCpBlkUnroll(GenTreeBlk* node)
             }
         }
 
+        // Handle the non-SIMD remainder by overlapping with previously processed data if needed
         if (size > 0)
         {
+            assert(size <= REGSIZE_BYTES);
+
+            // Round up to the closest power of two, but make sure it's not larger
+            // than the register we used for the main loop
+            regSize = min(regSize, roundUpGPRSize(size));
```

Comment:
Does this work correctly for x86 where not all registers can be addressed as byte sized?

Reply:
this path is for x64 only, yeah I had that in mind.